### ISSUE TITLE
Get-TestsFromBCContainer - The property 'Tests' cannot be found on this object.

### ISSUE THREAD

**Issue by sirhc101** [microsoft/navcontainerhelper#944]
Hi freddy,

hopefully you can help me with a problem. I've up and running 10+ build pipelines, based on your great blog posts regarding CI/CD. The pipelines all are working very well and running on a self hosted DevOps agent.
Now we started with a new product and cloned the scripts, created the build pipelines and got in trouble. Everything works fine, until it comes to step "Run Tests". There I receive:
```cmd
09:01:49.2346238Z ##[debug] Arguments: '-NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\agent\_work\_temp\40182dc3-14b0-4eea-9845-12df4ce3a02c.ps1'"'
09:01:49.2359248Z ##[debug] FileName: 'C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe'
09:01:49.2372068Z ##[debug] WorkingDirectory: 'C:\agent\_work\24\s'
09:01:49.2408445Z ##[command]"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\agent\_work\_temp\40182dc3-14b0-4eea-9845-12df4ce3a02c.ps1'"
09:02:00.6635002Z Open form
09:02:19.9560227Z Setting Extension Id 79081a2a-d3db-4c71-b144-af22b403e2e4
09:02:37.8715012Z Title: Command Line Test Tool Edit - Command Line Test Tool - DEFAULT
09:02:37.8735991Z Title: {0001fd97-0000-0001-0c00-0000836bd2d2}
09:02:37.8882183Z [
09:02:37.8886339Z     {
09:02:37.8890326Z         "type":  "ClientGroupControl",
09:02:37.8894295Z         "mappingHint":  "TOOLBAR",
09:02:37.8898320Z         "name":  "CommandBar",
09:02:37.8902239Z         "children":  [
09:02:37.8906298Z                          {
09:02:37.8910396Z                              "type":  "ClientGroupControl",
09:02:37.8914408Z                              "mappingHint":  "",
09:02:37.8918417Z                              "name":  "Control_Home",
09:02:37.8922817Z                              "children":  [
09:02:37.8926928Z                                               {
09:02:37.8931027Z                                                   "type":  "ClientGroupControl",
09:02:37.8934989Z                                                   "mappingHint":  "",
09:02:37.8938959Z                                                   "name":  "Control_Manage",
09:02:37.8942975Z                                                   "children":  [
09:02:37.8946961Z                                                                    {
09:02:37.8950901Z                                                                        "caption":  "Delete",
09:02:37.8954846Z                                                                        "type":  "ClientActionControl",
09:02:37.8958845Z                                                                        "name":  "Control_Delete",
09:02:37.8962827Z                                                                        "identifier":  "{0FCE965D-436A-418c-B11C-5D6EC3B94F3A}"
09:02:37.8966679Z                                                                    }
09:02:37.8970615Z                                                                ],
09:02:37.8974598Z                                                   "caption":  "Manage",
09:02:37.8979130Z                                                   "identifier":  "{37F878F7-1EB4-4B40-A6D5-55E87F26029B}"
09:02:37.8983803Z                                               },
09:02:37.8988267Z                                               {
09:02:37.8992532Z                                                   "type":  "ClientGroupControl",
09:02:37.8997583Z                                                   "mappingHint":  "",
09:02:37.9001986Z                                                   "name":  "Process",
09:02:37.9007053Z                                                   "children":  [
09:02:37.9012222Z                                                                    {
09:02:37.9016735Z                                                                        "caption":  "Run Se\u0026lected Tests",
09:02:37.9021149Z                                                                        "type":  "ClientActionControl",
09:02:37.9025481Z                                                                        "name":  "RunSelectedTests",
09:02:37.9029820Z                                                                        "identifier":  "{0001fd97-f33a-0000-0c50-3d00836bd2d2}"
09:02:37.9033951Z                                                                    },
09:02:37.9038766Z                                                                    {
09:02:37.9042981Z                                                                        "caption":  "Run N\u0026ext Test",
09:02:37.9047344Z                                                                        "type":  "ClientActionControl",
09:02:37.9051482Z                                                                        "name":  "RunNextTest",
09:02:37.9055811Z                                                                        "identifier":  "{0001fd97-8ad3-0000-0c18-8e00836bd2d2}"
09:02:37.9059998Z                                                                    },
09:02:37.9064268Z                                                                    {
09:02:37.9068481Z                                                                        "caption":  "Clear Test R\u0026esults",
09:02:37.9072559Z                                                                        "type":  "ClientActionControl",
09:02:37.9076887Z                                                                        "name":  "ClearTestResults",
09:02:37.9081026Z                                                                        "identifier":  "{0001fd97-2e16-0000-0c1a-0600836bd2d2}"
09:02:37.9085171Z                                                                    }
09:02:37.9089301Z                                                                ],
09:02:37.9093570Z                                                   "caption":  "Process",
09:02:37.9097799Z                                                   "identifier":  "{47F878F7-1EB4-4B40-A6D5-55E87F26029B}"
09:02:37.9193513Z                                               },
09:02:37.9202528Z                                               {
09:02:37.9207033Z                                                   "type":  "ClientGroupControl",
09:02:37.9211543Z                                                   "mappingHint":  "",
09:02:37.9215699Z                                                   "name":  "Page",
09:02:37.9219938Z                                                   "children":  [
09:02:37.9224040Z                                                                    {
09:02:37.9228464Z                                                                        "caption":  "Refresh",
09:02:37.9232568Z                                                                        "type":  "ClientActionControl",
09:02:37.9236641Z                                                                        "name":  "Control_Refresh",
09:02:37.9240890Z                                                                        "identifier":  "{57b9dd2f-e826-4d69-a433-34f779a5552d}"
09:02:37.9244895Z                                                                    }
09:02:37.9248930Z                                                                ],
09:02:37.9253181Z                                                   "caption":  "Page",
09:02:37.9257304Z                                                   "identifier":  "{CDBA910D-E283-4A38-AD03-C9D42763C3CE}"
09:02:37.9261387Z                                               }
09:02:37.9265496Z                                           ],
09:02:37.9269739Z                              "caption":  "Home",
09:02:37.9273906Z                              "identifier":  "{17F878F7-1EB4-4B40-A6D5-55E87F26029B}"
09:02:37.9278036Z                          },
09:02:37.9282195Z                          {
09:02:37.9286504Z                              "type":  "ClientGroupControl",
09:02:37.9290647Z                              "mappingHint":  "",
09:02:37.9296585Z                              "name":  "Control_Actions",
09:02:37.9305742Z                              "children":  [
09:02:37.9311984Z                                               {
09:02:37.9317272Z                                                   "caption":  "Run Se\u0026lected Tests",
09:02:37.9321543Z                                                   "type":  "ClientActionControl",
09:02:37.9326599Z                                                   "name":  "RunSelectedTests",
09:02:37.9331254Z                                                   "identifier":  "{0001fd97-f33a-0000-0c50-3d00836bd2d2}"
09:02:37.9336318Z                                               },
09:02:37.9341114Z                                               {
09:02:37.9345511Z                                                   "caption":  "Run N\u0026ext Test",
09:02:37.9349824Z                                                   "type":  "ClientActionControl",
09:02:37.9354214Z                                                   "name":  "RunNextTest",
09:02:37.9358404Z                                                   "identifier":  "{0001fd97-8ad3-0000-0c18-8e00836bd2d2}"
09:02:37.9363141Z                                               },
09:02:37.9367305Z                                               {
09:02:37.9371410Z                                                   "caption":  "Clear Test R\u0026esults",
09:02:37.9375254Z                                                   "type":  "ClientActionControl",
09:02:37.9379170Z                                                   "name":  "ClearTestResults",
09:02:37.9383052Z                                                   "identifier":  "{0001fd97-2e16-0000-0c1a-0600836bd2d2}"
09:02:37.9386906Z                                               }
09:02:37.9390828Z                                           ],
09:02:37.9394752Z                              "caption":  "\u0026Actions",
09:02:37.9399126Z                              "identifier":  "{67ff59f0-c388-4aab-8026-104dc2a9f53d}"
09:02:37.9406167Z                          },
09:02:37.9411622Z                          {
09:02:37.9417693Z                              "type":  "ClientGroupControl",
09:02:37.9424246Z                              "mappingHint":  "PAGEVIEWS",
09:02:37.9430150Z                              "name":  "ViewsGroup",
09:02:37.9434172Z                              "children":  [
09:02:37.9436179Z 
09:02:37.9440078Z                                           ],
09:02:37.9443751Z                              "caption":  "Views",
09:02:37.9447372Z                              "identifier":  ""
09:02:37.9450958Z                          },
09:02:37.9454904Z                          {
09:02:37.9458944Z                              "type":  "ClientGroupControl",
09:02:37.9463051Z                              "mappingHint":  "CommandBarHelpGroup",
09:02:37.9466899Z                              "name":  "HelpGroup",
09:02:37.9470770Z                              "children":  [
09:02:37.9472677Z 
09:02:37.9476631Z                                           ],
09:02:37.9480470Z                              "caption":  "",
09:02:37.9484273Z                              "identifier":  ""
09:02:37.9488021Z                          },
09:02:37.9491804Z                          {
09:02:37.9495871Z                              "type":  "ClientGroupControl",
09:02:37.9500335Z                              "mappingHint":  "CommandBarLayoutGroup",
09:02:37.9504225Z                              "name":  "LayoutGroup",
09:02:37.9508088Z                              "children":  [
09:02:37.9509989Z 
09:02:37.9513900Z                                           ],
09:02:37.9517963Z                              "caption":  "",
09:02:37.9521919Z                              "identifier":  ""
09:02:37.9525701Z                          },
09:02:37.9529588Z                          {
09:02:37.9533581Z                              "type":  "ClientStaticStringControl",
09:02:37.9537487Z                              "name":  "SystemIndicator",
09:02:37.9541470Z                              "value":  "",
09:02:37.9545414Z                              "identifier":  ""
09:02:37.9549290Z                          },
09:02:37.9553164Z                          {
09:02:37.9557111Z                              "type":  "ClientGroupControl",
09:02:37.9561045Z                              "mappingHint":  "",
09:02:37.9564947Z                              "name":  "Control_Microsoft_Dynamics_NAV",
09:02:37.9568772Z                              "children":  [
09:02:37.9573105Z                                               {
09:02:37.9579440Z                                                   "type":  "ClientGroupControl",
09:02:37.9583776Z                                                   "mappingHint":  "",
09:02:37.9588085Z                                                   "name":  "Page",
09:02:37.9592381Z                                                   "children":  [
09:02:37.9594428Z 
09:02:37.9598906Z                                                                ],
09:02:37.9603088Z                                                   "caption":  "Page",
09:02:37.9608684Z                                                   "identifier":  "{CDAB913B-E283-4A38-AD03-C9D42763C1CA}"
09:02:37.9612431Z                                               },
09:02:37.9616246Z                                               {
09:02:37.9620143Z                                                   "type":  "ClientGroupControl",
09:02:37.9623953Z                                                   "mappingHint":  "",
09:02:37.9627778Z                                                   "name":  "Control_Send_To",
09:02:37.9631677Z                                                   "children":  [
09:02:37.9633465Z 
09:02:37.9637571Z                                                                ],
09:02:37.9641570Z                                                   "caption":  "Print \u0026 Send",
09:02:37.9645576Z                                                   "identifier":  "{CDBB910B-E283-4A38-AD03-C9D42763C1CE}"
09:02:37.9649351Z                                               }
09:02:37.9653863Z                                           ],
09:02:37.9657852Z                              "caption":  "Dynamics 365 Business Central",
09:02:37.9661818Z                              "identifier":  "{d7c9f73f-9844-47ff-b4f4-88a77eb4e103}"
09:02:37.9665578Z                          }
09:02:37.9669537Z                      ],
09:02:37.9673384Z         "caption":  "",
09:02:37.9677264Z         "identifier":  "{FB8CCCF3-54AC-4d77-9AD8-136CEC25583A}"
09:02:37.9680968Z     },
09:02:37.9684740Z     {
09:02:37.9688661Z         "identifier":  "{886D4A84-3808-4545-92CF-374593FBC32F}",
09:02:37.9692422Z         "name":  "FilterControl",
09:02:37.9696263Z         "type":  "ClientFilterLogicalControl"
09:02:37.9699998Z     },
09:02:37.9703792Z     {
09:02:37.9707692Z         "type":  "ClientStringControl",
09:02:37.9714594Z         "name":  "CurrentSuiteName",
09:02:37.9721732Z         "value":  "DEFAULT",
09:02:37.9728339Z         "identifier":  "{0001fd97-e68d-0000-0c7f-8800836bd2d2}"
09:02:37.9732426Z     },
09:02:37.9736269Z     {
09:02:37.9740225Z         "type":  "ClientStringControl",
09:02:37.9743993Z         "name":  "TestCodeunitRangeFilter",
09:02:37.9747708Z         "value":  "",
09:02:37.9751632Z         "identifier":  "{0001fd97-c7e7-0000-0c54-2600836bd2d2}"
09:02:37.9755320Z     },
09:02:37.9759045Z     {
09:02:37.9762831Z         "type":  "ClientInt32Control",
09:02:37.9766555Z         "name":  "TestRunnerCodeunitId",
09:02:37.9770231Z         "value":  0,
09:02:37.9774132Z         "identifier":  "{0001fd97-cd7d-0000-0c44-3000836bd2d2}"
09:02:37.9777812Z     },
09:02:37.9781528Z     {
09:02:37.9785284Z         "type":  "ClientStringControl",
09:02:37.9788980Z         "name":  "ExtensionId",
09:02:37.9792724Z         "value":  "79081a2a-d3db-4c71-b144-af22b403e2e4",
09:02:37.9796544Z         "identifier":  "{0001fd97-70df-0000-0c66-d200836bd2d2}"
09:02:37.9800161Z     },
09:02:37.9803841Z     {
09:02:37.9807607Z         "type":  "ClientStringControl",
09:02:37.9811544Z         "name":  "DisableTestMethod",
09:02:37.9815239Z         "value":  "",
09:02:37.9819112Z         "identifier":  "{0001fd97-2386-0000-0c78-9900836bd2d2}"
09:02:37.9822790Z     },
09:02:37.9826496Z     {
09:02:37.9830482Z         "type":  "ClientStringControl",
09:02:37.9834566Z         "name":  "TestResultJson",
09:02:37.9838328Z         "value":  "",
09:02:37.9842213Z         "identifier":  "{0001fd97-5f9c-0000-0c69-0500836bd2d2}"
09:02:37.9845991Z     },
09:02:37.9849962Z     {
09:02:37.9853698Z         "Control1":  [
09:02:37.9857497Z                          {
09:02:37.9861313Z                              "":  null
09:02:37.9865084Z                          },
09:02:37.9868981Z                          {
09:02:37.9872853Z                              "":  null
09:02:37.9876652Z                          },
09:02:37.9880440Z                          {
09:02:37.9887882Z                              "":  null
09:02:37.9894230Z                          },
09:02:37.9898650Z                          {
09:02:37.9902357Z                              "":  null
09:02:37.9905981Z                          },
09:02:37.9909673Z                          {
09:02:37.9913400Z                              "":  null
09:02:37.9917075Z                          },
09:02:37.9920904Z                          {
09:02:37.9925208Z                              "":  null
09:02:37.9932005Z                          },
09:02:37.9975335Z                          {
09:02:37.9989810Z                              "":  null
09:02:37.9995930Z                          },
09:02:38.0000618Z                          {
09:02:38.0004970Z                              "":  null
09:02:38.0009177Z                          },
09:02:38.0013395Z                          {
09:02:38.0017627Z                              "":  null
09:02:38.0021791Z                          },
09:02:38.0025990Z                          {
09:02:38.0029940Z                              "":  null
09:02:38.0034985Z                          },
09:02:38.0039747Z                          {
09:02:38.0043940Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0047980Z                              "TestCodeunit":  62030,
09:02:38.0052495Z                              "LineType":  "0",
09:02:38.0057013Z                              "Run":  true,
09:02:38.0061384Z                              "Name":  "",
09:02:38.0065600Z                              "ErrorMessage":  "",
09:02:38.0070003Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0073454Z                              "Result":  "0",
09:02:38.0077048Z                              "StackTrace":  ""
09:02:38.0080553Z                          },
09:02:38.0084123Z                          {
09:02:38.0087788Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0091299Z                              "TestCodeunit":  62030,
09:02:38.0094857Z                              "LineType":  "",
09:02:38.0098419Z                              "Run":  true,
09:02:38.0101932Z                              "Name":  "",
09:02:38.0105496Z                              "ErrorMessage":  "",
09:02:38.0109096Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0112594Z                              "Result":  "",
09:02:38.0116131Z                              "StackTrace":  ""
09:02:38.0119629Z                          },
09:02:38.0123641Z                          {
09:02:38.0128054Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0131833Z                              "TestCodeunit":  62030,
09:02:38.0135542Z                              "LineType":  "1",
09:02:38.0139153Z                              "Run":  true,
09:02:38.0142928Z                              "Name":  "TestValidateContactVerificationDisabled",
09:02:38.0146417Z                              "ErrorMessage":  "",
09:02:38.0150150Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0153624Z                              "Result":  "0",
09:02:38.0157272Z                              "StackTrace":  ""
09:02:38.0161123Z                          },
09:02:38.0165187Z                          {
09:02:38.0168988Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0172596Z                              "TestCodeunit":  0,
09:02:38.0176237Z                              "LineType":  "0",
09:02:38.0180090Z                              "Run":  true,
09:02:38.0183743Z                              "Name":  "",
09:02:38.0187362Z                              "ErrorMessage":  "",
09:02:38.0191034Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0194580Z                              "Result":  "0",
09:02:38.0198401Z                              "StackTrace":  ""
09:02:38.0202407Z                          },
09:02:38.0206155Z                          {
09:02:38.0209866Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0213536Z                              "TestCodeunit":  0,
09:02:38.0217668Z                              "LineType":  "",
09:02:38.0221887Z                              "Run":  false,
09:02:38.0225712Z                              "Name":  "",
09:02:38.0229535Z                              "ErrorMessage":  "",
09:02:38.0233351Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0237096Z                              "Result":  "",
09:02:38.0240859Z                              "StackTrace":  ""
09:02:38.0244633Z                          },
09:02:38.0248379Z                          {
09:02:38.0252231Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0255904Z                              "TestCodeunit":  62030,
09:02:38.0259655Z                              "LineType":  "0",
09:02:38.0263411Z                              "Run":  true,
09:02:38.0267130Z                              "Name":  "",
09:02:38.0274345Z                              "ErrorMessage":  "",
09:02:38.0274788Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0278367Z                              "Result":  "0",
09:02:38.0282212Z                              "StackTrace":  ""
09:02:38.0285928Z                          },
09:02:38.0289690Z                          {
09:02:38.0293495Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0297219Z                              "TestCodeunit":  62030,
09:02:38.0300947Z                              "LineType":  "",
09:02:38.0304663Z                              "Run":  true,
09:02:38.0308390Z                              "Name":  "",
09:02:38.0312142Z                              "ErrorMessage":  "",
09:02:38.0315903Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0319684Z                              "Result":  "",
09:02:38.0323443Z                              "StackTrace":  ""
09:02:38.0327154Z                          },
09:02:38.0330899Z                          {
09:02:38.0334691Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0338404Z                              "TestCodeunit":  62030,
09:02:38.0342136Z                              "LineType":  "1",
09:02:38.0345834Z                              "Run":  true,
09:02:38.0349878Z                              "Name":  "TestValidateContactVerificationDisabled",
09:02:38.0353594Z                              "ErrorMessage":  "",
09:02:38.0357349Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0361067Z                              "Result":  "0",
09:02:38.0364783Z                              "StackTrace":  ""
09:02:38.0368477Z                          },
09:02:38.0372214Z                          {
09:02:38.0376009Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0380132Z                              "TestCodeunit":  0,
09:02:38.0383943Z                              "LineType":  "0",
09:02:38.0387709Z                              "Run":  true,
09:02:38.0391423Z                              "Name":  "",
09:02:38.0395186Z                              "ErrorMessage":  "",
09:02:38.0399334Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0403099Z                              "Result":  "0",
09:02:38.0406870Z                              "StackTrace":  ""
09:02:38.0410577Z                          },
09:02:38.0414331Z                          {
09:02:38.0418151Z                              "FinishTime":  "\/Date(-62135596800000)\/",
09:02:38.0421877Z                              "TestCodeunit":  0,
09:02:38.0425609Z                              "LineType":  "",
09:02:38.0429347Z                              "Run":  false,
09:02:38.0433074Z                              "Name":  "",
09:02:38.0437076Z                              "ErrorMessage":  "",
09:02:38.0440879Z                              "StartTime":  "\/Date(-62135596800000)\/",
09:02:38.0444593Z                              "Result":  "",
09:02:38.0448356Z                              "StackTrace":  ""
09:02:38.0452053Z                          }
09:02:38.0455785Z                      ],
09:02:38.0460031Z         "type":  "ClientRepeaterControl",
09:02:38.0463818Z         "name":  "Control1",
09:02:38.0467670Z         "identifier":  "{0001fd97-2e1d-0000-0c0b-2000836bd2d2}"
09:02:38.0471315Z     },
09:02:38.0475091Z     {
09:02:38.0478875Z         "type":  "ClientGroupControl",
09:02:38.0482611Z         "mappingHint":  "ACTIONBAR",
09:02:38.0486321Z         "name":  "ActionBar",
09:02:38.0490046Z         "children":  [
09:02:38.0493782Z                          {
09:02:38.0497553Z                              "caption":  "OK",
09:02:38.0501189Z                              "type":  "ClientActionControl",
09:02:38.0504924Z                              "name":  "{B594146A-9CEF-4767-AE76-E9721CD6FB50}",
09:02:38.0508652Z                              "identifier":  "{B594146A-9CEF-4767-AE76-E9721CD6FB50}"
09:02:38.0512273Z                          }
09:02:38.0515978Z                      ],
09:02:38.0519683Z         "caption":  "",
09:02:38.0523461Z         "identifier":  "{EC1DA4B5-D7A8-4091-A7B3-2DB6107B6C56}"
09:02:38.0527084Z     }
09:02:38.0530796Z ]
09:02:38.2755736Z The property 'Tests' cannot be found on this object. Verify that the property exists and can be set.
09:02:38.2779155Z at Get-Tests, C:\ProgramData\NavContainerHelper\Extensions\businessdev-srv02-build\PsTestTool-4\PsTestFunctions.ps1: line 218
09:02:38.2791653Z at <ScriptBlock>, <No file>: line 51
09:02:38.3797383Z ##[debug]STDERR: The property 'Tests' cannot be found on this object. Verify that the property exists and can be set.
09:02:38.3822780Z ##[debug]STDERR: At C:\Program 
09:02:38.3831506Z ##[debug]
09:02:38.3847856Z ##[debug]STDERR: Files\WindowsPowerShell\Modules\navcontainerhelper\0.6.5.6\ContainerHandling\Invoke-ScriptInNavContainer.ps1:37 char:13
09:02:38.3858780Z ##[debug]
09:02:38.3875411Z ##[debug]STDERR: +             Invoke-Command -Session $session -ScriptBlock $scriptbloc ...
09:02:38.3884134Z ##[debug]
09:02:38.3948781Z ##[debug]STDERR: +             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
09:02:38.3950422Z ##[debug]
09:02:38.3951300Z ##[debug]STDERR:     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
09:02:38.3952008Z ##[debug]
09:02:38.3966468Z ##[debug]STDERR:     + FullyQualifiedErrorId : PropertyNotFound
09:02:38.3978005Z ##[debug]
09:02:38.3993590Z ##[debug]STDERR:  
09:02:38.4001794Z ##[debug]
09:02:38.4017619Z ##[debug]Exit code: 1
```

I tried to figure out whats wrong for multiple days now, but I've no idea anymore so hopefully you can help me out.

What I've already tried is
- Remove all NavContainerHelper version
- Reinstall newest version
- Re-create build pipeline / cloned newly from a running pipeline

The weird thing is, when I run the tests in web client (also at the build container) everything works fine. When I run the tests using PowerShell and your "Run-Test.ps1" script everthing works fine.

The problem only appears when it's inside the pipeline.

**Comment by freddydk** [comment]
Are there any tests in the project?

**Comment by sirhc101** [comment]
Yes, of course. There are to AL project folders, called app and test.
app does not contain any test codeunits, but test does.
The extension ID you see in the output is the appId of the test app, so everything looks good.

**Comment by sirhc101** [comment]
Maybe this is important.

I've no found the exact code where the exception will be thrown. It's this code in `PsTestFunctions.ps1`:
```ps
if ($codeunitId -like $testCodeunit -or $codeunitName -like $testCodeunit) {
  if ($run) {
    $codeunit.Tests += $name
  }
}
```
When I understand correctly what you try to do, than you iterate through the repeater control of page 130455 and depending on the `LineType` control you identify what the line is.
0 = Group
1 = Codeunit
2 = Test

I modifed the file, to write some more output, so it now looks like this:
```ps
        if ($name) {
            if ($linetype -eq "0" -and !$ignoreGroups) {
		if ($debugMode) {
			Write-Host "Found Group $name"
		}
                $group = @{ "Group" = $name; "Codeunits" = @() }
                $Tests += $group
                            
            } elseif ($linetype -eq "1") {
		if ($debugMode) {
			Write-Host " Found Codeunit $name"
		}
                $codeUnitName = $name
                if ($codeunitId -like $testCodeunit -or $codeunitName -like $testCodeunit) {
                    $codeunit = @{ "Id" = "$codeunitId"; "Name" = $codeUnitName; "Tests" = @() }
                    if ($group) {
                        $group.Codeunits += $codeunit
                    }
                    else {
                        if ($run) {
                            $Tests += $codeunit
                        }
                    }
                }
            } elseif ($lineType -eq "2") {
		if ($debugMode) {
			Write-Host "  Found Test $name"
		}
                if ($codeunitId -like $testCodeunit -or $codeunitName -like $testCodeunit) {
                    if ($run) {
                        $codeunit.Tests += $name
                    }
                }
            }
        }
```

As far as I understand your code the order must be at least first codeunit, than test procedure, right? The problem is, that the output is:
```cmd
"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\agent\_work\_temp\0d24cd44-7b4f-4d4c-a541-0a9a5c528de0.ps1'"
Open form 130455
Setting Extension Id 79081a2a-d3db-4c71-b144-af22b403e2e4
Searching Repeater...
  Found Test TestAutomaticAddressValidationEnabledUnsupportedTable
Title: Command Line Test Tool Edit - Command Line Test Tool - DEFAULT
Title: {0001fd97-0000-0001-0c00-0000836bd2d2}
[..]
```

The output of an working pipeline is looking the right way:
```cmd
"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\agent\_work\_temp\1f6f3d8e-b66b-4e33-ae34-585aac407e33.ps1'"
Open form 130455
Setting Extension Id 456b3cd5-7686-4038-89fb-573bfd481647
Searching Repeater...
 Found Codeunit TestCodeunit
  Found Test Test1
  Found Test Test2
  Found Test Test3
  Codeunit 62031 TestCodeunit (1.006 seconds)
```

If I open the command line test tool in web client the ordering appears correct:
![image](https://user-images.githubusercontent.com/39621236/79336618-5af86580-7f24-11ea-9474-7240b0d2ff70.png)

**Comment by sirhc101** [comment]
Ok, I figured out what's wrong. It looks like a problem in `PsTestFunctions.ps1` regarding handling with the offset if you've more than 15 tests.

Comming back after I completely solved it.

**Comment by sirhc101** [comment]
Hi freddy,

I solved the problem by modifying the PsTestFunctions.ps1. I tested it this night against all of my 13 build pipelines and it worked perfectly. I've no idea why it worked directly from the PowerShell ISE. Looks like the behaviour in the pipeline regarding the UI DLL is different than in the pipeline.

However, it now works on both.
I created the pull request #947.

**Comment by freddydk** [comment]
Thanks a lot.
Did your tests test both the AL Test Runner and the C/AL test runner?
Or are all your projects BC 15.x +?

**Comment by sirhc101** [comment]
> Did your tests test both the AL Test Runner and the C/AL test runner?
To be honest, I totally forgot the C/AL test runner. :-( 

> Or are all your projects BC 15.x +?
Yes, all projects are BC15+

**Comment by freddydk** [comment]
I will look through this, thanks.

**Comment by sirhc101** [comment]
Hi freddy,

Pls do not merge the pull request yet. I got a notice from one of my pipelines that the error occurred again.
Need to take a look into it...

**Comment by sirhc101** [comment]
@freddydk 
I've updated the pullrequest #947 again.

I've tested the new version against all my apps and some new one which I just created to test the pipelines and automated runs.

Pls double-check from your side to verify, but I think it should ne be fine.

**Comment by freddydk** [comment]
Looking at this now.
Could you maybe paste your run-tests script here? (I have probably have multiple versions of that, just want to see what parameters you use to call get-tests)
I am trying to find a repro-able scenario myself.

**Comment by sirhc101** [comment]
Sorry, I didn't noticed your reply.

Attached you find the Run-Test.ps1 I'm using in the pipeline where it crashed.
[Run-Tests.ps1.txt](https://github.com/microsoft/navcontainerhelper/files/4547248/Run-Tests.ps1.txt)

**Comment by freddydk** [comment]
Did you have a chance to test the fix I mentioned in the PR?

**Comment by freddydk** [comment]
Have checked in a fix for this just now - I ended up having a repro in some versions and other versions working. This latest fix works for me in current, next minor and next major.

**Comment by freddydk** [comment]
Shipped in 0.6.5.7
